plutus-core/untyped-plutus-core/test/Evaluation/Builtins/Conversion.hs

```diff
@@ -0,0 +1,459 @@
+
+-- indexByteString (builtinIntegerToByteString False d i) 0 = remainderInteger i 256
+i2bProperty3 :: PropertyT IO ()
+i2bProperty3 = do
+  (d, i) <- forAllWith ppShow genProp3Prop4Data
+  let expectedExp = mkIterAppNoAnn (builtin () RemainderInteger) [
+        mkConstant @Integer () i,
+        mkConstant @Integer () 256
+        ]
+  let actualExp = mkIterAppNoAnn (builtin () IntegerToByteString) [
+        mkConstant @Bool () False,
+        mkConstant @Integer () d,
+        mkConstant @Integer () i
+        ]
+  let indexExp = mkIterAppNoAnn (builtin () IndexByteString) [
+        actualExp,
+        mkConstant @Integer () 0
+        ]
+  let expectedResult = typecheckEvaluateCek def defaultBuiltinCostModel expectedExp
+  let actualResult = typecheckEvaluateCek def defaultBuiltinCostModel indexExp
+  case (expectedResult, actualResult) of
+    (Left err, _) -> annotateShow err >> failure
+    (_, Left err) -> annotateShow err >> failure
+    (Right (eRes, eLogs), Right (aRes, aLogs)) -> case (eRes, aRes) of
+      (EvaluationFailure, _)                                 -> annotateShow eLogs >> failure
+      (_, EvaluationFailure)                                 -> annotateShow aLogs >> failure
+      (EvaluationSuccess eResult, EvaluationSuccess aResult) -> eResult === aResult
+
+-- let result = builtinIntegerToByteString True d i
+-- in indexByteString result (lengthOfByteString result - 1) = remainderInteger i 256
+i2bProperty4 :: PropertyT IO ()
+i2bProperty4 = do
+  (d, i) <- forAllWith ppShow genProp3Prop4Data
+  let expectedExp = mkIterAppNoAnn (builtin () RemainderInteger) [
+        mkConstant @Integer () i,
+        mkConstant @Integer () 256
+        ]
+  let actualExp = mkIterAppNoAnn (builtin () IntegerToByteString) [
+        mkConstant @Bool () True,
+        mkConstant @Integer () d,
+        mkConstant @Integer () i
+        ]
+  let lenExp = mkIterAppNoAnn (builtin () LengthOfByteString) [
+        actualExp
+        ]
+  let limitExp = mkIterAppNoAnn (builtin () SubtractInteger) [
+        lenExp,
+        mkConstant @Integer () 1
+        ]
+  let indexExp = mkIterAppNoAnn (builtin () IndexByteString) [
+        actualExp,
+        limitExp
+        ]
+  let expectedResult = typecheckEvaluateCek def defaultBuiltinCostModel expectedExp
+  let actualResult = typecheckEvaluateCek def defaultBuiltinCostModel indexExp
+  case (expectedResult, actualResult) of
+    (Left err, _) -> annotateShow err >> failure
+    (_, Left err) -> annotateShow err >> failure
+    (Right (eRes, eLogs), Right (aRes, aLogs)) -> case (eRes, aRes) of
+      (EvaluationFailure, _)                                 -> annotateShow eLogs >> failure
+      (_, EvaluationFailure)                                 -> annotateShow aLogs >> failure
+      (EvaluationSuccess eResult, EvaluationSuccess aResult) -> eResult === aResult
+
+-- builtinByteStringToInteger b (builtinIntegerToByteString b d i) = i
```

Comment:
Can we check the round trip in the opposite direction, at least for width zero?

Reply:
Yes, that's definitely possible. Should add it to the CIP as well.